src/stackmap_checker/utils.c

```diff
@@ -20,9 +20,11 @@ char* get_binary_path()
     return buff;
 }
 
-Elf64_Ehdr* get_elf_header(const char *bin_name)
+Elf64_Ehdr* get_elf_header()
 {
+    char *bin_name = get_binary_path();
```

Comment:
There's a linux function (it's not portable) to get the path, which might avoid having this allocate. I don't remember the name, sorry.

Reply:
Ah, I see. I couldn't come up with a better solution at the time

Reply:
It's fine as is tbh. Just make a mental note for the next time you need this :)

Reply:
Ok, thank you for pointing it out!